frontend/device/input.lua

```diff
@@ -449,6 +458,29 @@ function Input:handleTouchEvPhoenix(ev)
     end
 end
 
+function Input:handleOrientationEv(ev)
+
+    if ev.type == EV_ABS then
+        if ev.code == ABS_PRESSURE and self.device.screen:getScreenMode() == 'portrait' then
+            local refreshUI = false
+            local rotation_mode = 0
+            if ev.value == DEVICE_ORIENTATION_PORTRAIT then
+                refreshUI = true
+            elseif ev.value == DEVICE_ORIENTATION_PORTRAIT_ROTATED then
+                refreshUI = true
+                rotation_mode = 2
+            end
+
+            local oldRotation = self.device.screen:getRotationMode()
+            if refreshUI and rotation_mode ~= oldRotation then
+                self.device.screen:setRotationMode(rotation_mode)
+                local UIManager = require("ui/uimanager")
+                UIManager:onRotation()
```

Comment:
The best way to do this is to return a event from this handler and let the UI loop handles it. But it will require refactoring some of the UI loop code. So you can leave it as is for now. I can help tidy it up after this PR is merged :)